sor/src/main/java/com/bazaarvoice/emodb/sor/db/DAOUtils.java

```diff
@@ -8,17 +8,24 @@
 import java.nio.ByteBuffer;
 import java.util.List;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
 public class DAOUtils {
 
     private final int _prefixLength;
     private final int _deltaBlockSize;
     private final byte[] _singleBlockBytes;
+    private final int _maxBlocks;
 
     @Inject
     public DAOUtils(@PrefixLength  int prefixLength, @BlockSize int deltaBlockSize) {
+        checkArgument(prefixLength > 0, "Prefix length must be greater than 0");
+        checkArgument(deltaBlockSize > 0, "Delta block size must be greater than 0");
+
         _prefixLength = prefixLength;
         _deltaBlockSize = deltaBlockSize;
         _singleBlockBytes = String.format("%0" + prefixLength + "X", 1).getBytes();
+        _maxBlocks = (int) Math.pow(16, _prefixLength) - 1;
```

Comment:
Just for my sake can you verify my understanding:  It's 16^_prefixLength because the prefix is hex-encoded, right?  Might be worth a comment since that connection isn't immediately clear in this context.

Reply:
yes. I will add a comment